src/SDiagonal.jl

```diff
@@ -17,6 +17,11 @@ SDiagonal(a::StaticMatrix{N,N,T}) where {N,T} = Diagonal(diag(a))
 size(::Type{<:SDiagonal{N}}) where {N} = (N,N)
 size(::Type{<:SDiagonal{N}}, d::Int) where {N} = d > 2 ? 1 : N
 
+Base.axes(D::SDiagonal) = (ax = axes(diag(D), 1); (ax, ax))
+Base.axes(D::SDiagonal, d) = d <= 2 ? axes(D)[d] : SOneTo(1)
```

Comment:
Unlike some cases identified in https://github.com/JuliaArrays/StaticArrays.jl/pull/916, there seems to be currently no type-instability here but this method introduces it.

Reply:
Yes, I think this is fine.